src/ViewModels/src/RemoteDebug/RemoteDebugViewModel.cs

```diff
@@ -413,16 +411,57 @@ public void DisplayDeploymentWindow(object arg = null)
             }
         }
 
-        private void PromptAppResolution(string promptMsg)
+        private void ResetState()
         {
+            AppToDebug = null;
+            LoadingMessage = null;
+            IsLoggedIn = _tasExplorer != null && _tasExplorer.TasConnection != null;
+            WaitingOnAppConfirmation = false;
+            CanCancel = true;
+            CancelDebugging = (object arg) =>
+            {
+                Close();
+            };
+        }
+
+        private async Task<bool> CheckForVsdbg(string stack, CancellationToken ct)
+        {
+            StopDebuggingAndCloseIfCancelled(ct);
+
+            LoadingMessage = $"Checking for debugging agent on {AppToDebug.AppName}...";
+            DetailedResult sshResult;
+            bool vsdbExecutableListed;
+
+            if (stack.Contains("win"))
+            {
+                var sshCommand = $"dir {_vsdbgInstallationDirWindows}";
+                sshResult = await _cfCliService.ExecuteSshCommand(AppToDebug.AppName, AppToDebug.ParentSpace.ParentOrg.OrgName, AppToDebug.ParentSpace.SpaceName, sshCommand);
+                vsdbExecutableListed = sshResult.CmdResult.StdOut.Contains(_vsdbgExecutableNameWindows);
+            }
+            else
+            {
+                var sshCommand = $"ls {_vsdbgInstallationDirLinux}";
+                sshResult = await _cfCliService.ExecuteSshCommand(AppToDebug.AppName, AppToDebug.ParentSpace.ParentOrg.OrgName, AppToDebug.ParentSpace.SpaceName, sshCommand);
+                vsdbExecutableListed = sshResult.CmdResult.StdOut.Contains(_vsdbgExecutableNameLinux);
+            }
```

Comment:
It looks like there is little difference between these two blocks. It might be easier to reason about for a future reader if the duplicated lines are moved after the if block, something like this:
```suggestion
            string sshCommand;
            string vsdbg;
            if (stack.Contains("win"))
            {
                sshCommand = $"dir {_vsdbgInstallationDirWindows}";
                vsdbg = _vsdbgExecutableNameWindows;
            }
            else
            {
                sshCommand = $"ls {_vsdbgInstallationDirLinux}";
                vsdbg = _vsdbgExecutableNameLinux;
            }
            
            sshResult = await _cfCliService.ExecuteSshCommand(AppToDebug.AppName, AppToDebug.ParentSpace.ParentOrg.OrgName, AppToDebug.ParentSpace.SpaceName, sshCommand);
            vsdbExecutableListed = sshResult.CmdResult.StdOut.Contains(_vsdbgExecutableNameWindows);
```
A further optimization would be to pick either linux or windows as the default and then you could drop the `else` block as well. Please note this suggestion was typed out on github and hasn't been compiler-checked

Reply:
👌 love it, yep this whole file could use a thorough revisit for simplicity tbh... Great find for a quick improvement though! Will take a crack at incorporating it

Reply:
Done in [df89715](https://github.com/vmware-tanzu/tanzu-toolkit-for-visual-studio/pull/354/commits/df897150f2cdda391c0ebd950a08b992943307e2)